pkg/tls/cred.go

```diff
@@ -0,0 +1,186 @@
+package tls
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+)
+
+type (
+	// Cred is a container for a certificate, trust chain, and private key.
+	Cred struct {
+		PrivateKey *ecdsa.PrivateKey
+		Crt
+	}
+
+	// Crt is a container for a certificate and trust chain.
+	//
+	// The trust chain stores all issuer certificates from root at the head direct
+	// issuer at the tail.
```

Comment:
This might be more readable as:

```
	// The trust chain stores all issuer certificates from the root at the head to
	// the direct issuer at the tail.
```